Content.Server/Chat/TelepathicChatSystem.Psychognomy.cs

```diff
@@ -0,0 +1,187 @@
+        {
+            if (component.CurrentState == Shared.Mobs.MobState.Critical)
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-liminal"));
+        }
+
+        private void DescribeHunger(EntityUid uid, HungerComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.CurrentThreshold <= HungerThreshold.Peckish)
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-hungry"));
+        }
+
+        private void DescribeFixtures(EntityUid uid, FixturesComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            foreach (var fixture in component.Fixtures.Values)
+            {
+                if (fixture.CollisionMask == (int) CollisionGroup.GhostImpassable)
+                {
+                    ev.Descriptors.Add(Loc.GetString("p-descriptor-pneumatic"));
+                    return;
+                }
+            }
+        }
+
+        private void DescribePhysics(EntityUid uid, PhysicsComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.FixturesMass < 45)
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-light"));
+            else if (component.FixturesMass > 70)
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-heavy"));
+        }
+
+        private void DescribeKarma(EntityUid uid, MetempsychosisKarmaComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.Score > 0)
+            {
+                ev.Descriptors.Add(Loc.GetString("cyclic"));
+            }
+        }
+
+        private void DescribeGlimmerSource(EntityUid uid, GlimmerSourceComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (component.AddToGlimmer)
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-emanative"));
+            else
+            {
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-vampiric"));
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-hungry"));
+            }
+        }
+
+        // This one's also a bit of a catch-all for "lacks component"
+        private void DescribePsion(EntityUid uid, PsionicComponent component, GetPsychognomicDescriptorEvent ev)
+        {
+            if (!HasComp<SpeechComponent>(uid) || HasComp<MutedComponent>(uid))
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-dumb"));
+
+            if (!HasComp<CombatModeComponent>(uid) || HasComp<PacifiedComponent>(uid))
+                ev.Descriptors.Add(Loc.GetString("p-descriptor-passive"));
+
+            foreach (var power in component.ActivePowers)
+            {
+                // TODO: Mime counts too and isn't added back to psions yet
+                if (power.ID == "PyrokinesisPower" || power.ID == "NoosphericZapPower")
+                    ev.Descriptors.Add(Loc.GetString("p-descriptor-kinetic"));
+
+                return;
+            }
```

Comment:
```suggestion
            {
                // TODO: Mime counts too and isn't added back to psions yet
                if (power.ID != "PyrokinesisPower" && power.ID != "NoosphericZapPower")
                    continue;

                ev.Descriptors.Add(Loc.GetString("p-descriptor-kinetic"));
                return;
            }
```
This one was a logic error. It would check the first power in the list, and if it wasn't Pyrokinesis or Noospheric Zap, stop checking the list and exit the function.